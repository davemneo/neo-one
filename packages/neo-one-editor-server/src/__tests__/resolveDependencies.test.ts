import { resolveDependencies } from '../resolveDependencies';

const dependencies = {
  'bignumber.js': '7.2.1',
  react: '16.5.1',
  reakit: '0.15.7',
  rxjs: '6.3.3',
  'react-dom': '16.5.1',
  'styled-components': '3.4.10',
};

describe('resolveDependencies', () => {
  test('resolve deps', async () => {
    const resolvedDependencies = await resolveDependencies(dependencies);

    expect(resolvedDependencies).toMatchSnapshot();
  });
});
