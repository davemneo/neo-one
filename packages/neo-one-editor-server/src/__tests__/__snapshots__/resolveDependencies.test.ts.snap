// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolveDependencies resolve deps 1`] = `
Object {
  "@emotion/is-prop-valid": Object {
    "dependencies": Object {},
    "name": "@emotion/is-prop-valid",
    "types": false,
    "version": "0.6.8",
  },
  "@emotion/memoize": Object {
    "dependencies": Object {},
    "name": "@emotion/memoize",
    "types": false,
    "version": "0.6.6",
  },
  "asap": Object {
    "dependencies": Object {},
    "name": "asap",
    "types": false,
    "version": "2.0.6",
  },
  "base64-js": Object {
    "dependencies": Object {},
    "name": "base64-js",
    "types": false,
    "version": "1.3.0",
  },
  "bignumber.js": Object {
    "dependencies": Object {},
    "name": "bignumber.js",
    "types": true,
    "version": "7.2.1",
  },
  "buffer": Object {
    "dependencies": Object {},
    "name": "buffer",
    "types": true,
    "version": "5.2.1",
  },
  "constate": Object {
    "dependencies": Object {},
    "name": "constate",
    "types": true,
    "version": "0.9.0",
  },
  "core-js": Object {
    "dependencies": Object {},
    "name": "core-js",
    "types": false,
    "version": "1.2.7",
  },
  "css-color-keywords": Object {
    "dependencies": Object {},
    "name": "css-color-keywords",
    "types": false,
    "version": "1.0.0",
  },
  "css-to-react-native": Object {
    "dependencies": Object {},
    "name": "css-to-react-native",
    "types": false,
    "version": "2.2.2",
  },
  "encoding": Object {
    "dependencies": Object {},
    "name": "encoding",
    "types": false,
    "version": "0.1.12",
  },
  "fbjs": Object {
    "dependencies": Object {},
    "name": "fbjs",
    "types": false,
    "version": "0.8.17",
  },
  "has-flag": Object {
    "dependencies": Object {},
    "name": "has-flag",
    "types": false,
    "version": "1.0.0",
  },
  "hoist-non-react-statics": Object {
    "dependencies": Object {},
    "name": "hoist-non-react-statics",
    "types": true,
    "version": "2.5.5",
  },
  "iconv-lite": Object {
    "dependencies": Object {},
    "name": "iconv-lite",
    "types": true,
    "version": "0.4.24",
  },
  "ieee754": Object {
    "dependencies": Object {},
    "name": "ieee754",
    "types": false,
    "version": "1.1.12",
  },
  "is-stream": Object {
    "dependencies": Object {},
    "name": "is-stream",
    "types": false,
    "version": "1.1.0",
  },
  "isomorphic-fetch": Object {
    "dependencies": Object {},
    "name": "isomorphic-fetch",
    "types": false,
    "version": "2.2.1",
  },
  "js-tokens": Object {
    "dependencies": Object {},
    "name": "js-tokens",
    "types": false,
    "version": "4.0.0",
  },
  "loose-envify": Object {
    "dependencies": Object {},
    "name": "loose-envify",
    "types": false,
    "version": "1.4.0",
  },
  "node-fetch": Object {
    "dependencies": Object {},
    "name": "node-fetch",
    "types": false,
    "version": "1.7.3",
  },
  "object-assign": Object {
    "dependencies": Object {},
    "name": "object-assign",
    "types": false,
    "version": "4.1.1",
  },
  "popper.js": Object {
    "dependencies": Object {},
    "name": "popper.js",
    "types": true,
    "version": "1.14.5",
  },
  "postcss-value-parser": Object {
    "dependencies": Object {},
    "name": "postcss-value-parser",
    "types": false,
    "version": "3.3.1",
  },
  "promise": Object {
    "dependencies": Object {},
    "name": "promise",
    "types": false,
    "version": "7.3.1",
  },
  "prop-types": Object {
    "dependencies": Object {},
    "name": "prop-types",
    "types": false,
    "version": "15.6.2",
  },
  "react": Object {
    "dependencies": Object {},
    "name": "react",
    "types": false,
    "version": "16.5.1",
  },
  "react-dom": Object {
    "dependencies": Object {},
    "name": "react-dom",
    "types": false,
    "version": "16.5.1",
  },
  "react-is": Object {
    "dependencies": Object {},
    "name": "react-is",
    "types": false,
    "version": "16.6.2",
  },
  "reakit": Object {
    "dependencies": Object {},
    "name": "reakit",
    "types": true,
    "version": "0.15.7",
  },
  "rxjs": Object {
    "dependencies": Object {},
    "name": "rxjs",
    "types": true,
    "version": "6.3.3",
  },
  "safer-buffer": Object {
    "dependencies": Object {},
    "name": "safer-buffer",
    "types": false,
    "version": "2.1.2",
  },
  "schedule": Object {
    "dependencies": Object {},
    "name": "schedule",
    "types": false,
    "version": "0.4.0",
  },
  "setimmediate": Object {
    "dependencies": Object {},
    "name": "setimmediate",
    "types": false,
    "version": "1.0.5",
  },
  "styled-components": Object {
    "dependencies": Object {},
    "name": "styled-components",
    "types": true,
    "version": "3.4.10",
  },
  "styled-tools": Object {
    "dependencies": Object {},
    "name": "styled-tools",
    "types": true,
    "version": "1.6.0",
  },
  "stylis": Object {
    "dependencies": Object {},
    "name": "stylis",
    "types": true,
    "version": "3.5.4",
  },
  "stylis-rule-sheet": Object {
    "dependencies": Object {},
    "name": "stylis-rule-sheet",
    "types": false,
    "version": "0.0.10",
  },
  "supports-color": Object {
    "dependencies": Object {},
    "name": "supports-color",
    "types": false,
    "version": "3.2.3",
  },
  "tslib": Object {
    "dependencies": Object {},
    "name": "tslib",
    "types": true,
    "version": "1.9.3",
  },
  "ua-parser-js": Object {
    "dependencies": Object {},
    "name": "ua-parser-js",
    "types": false,
    "version": "0.7.19",
  },
  "whatwg-fetch": Object {
    "dependencies": Object {},
    "name": "whatwg-fetch",
    "types": false,
    "version": "3.0.0",
  },
}
`;
